### PR TITLE
test: Add GitHub Actions test support

### PR DESCRIPTION
**Pull request by frebib**
Contains some other small cleanups, and some necessary improvements to make the CI formatting a bit tidier and cleaner.

### CODE REVIEW

**Comment by frebib** [comment]
I can't request reviewers @purpleidea 

cc @evrardjp based on your work

**Comment by frebib** [comment]
Seems the test-shell is failing. I've not looked into why yet but I presume it's either a flaky test or the test is broken
https://github.com/frebib/mgmt/runs/1889240885?check_suite_focus=true#step:6:129

**Comment by frebib** [comment]
Whilst I think about it (more of a note to myself more than anything) we need a step to send an IRC message the same way that Travis does

**Comment by frebib** [comment]
> Seems the test-shell is failing. I've not looked into why yet but I presume it's either a flaky test or the test is broken
> https://github.com/frebib/mgmt/runs/1889240885?check_suite_focus=true#step:6:129

I'm starting to think that these shell tests are a bit flaky. I ran them locally and the one that failed in CI passed fine locally, but a couple of others later on failed for unrelated and silly reasons (couldn't bind, port already in use)

File: test/util.sh [hunk 1]
```diff
@@ -20,26 +20,43 @@ fi
...
 	"$@" || failures=$( [ -n "$failures" ] && echo "$failures\\n$@" || echo "$@" )
 }
 
 # travis expander helpers from:
 # https://github.com/travis-ci/travis-rubies/blob/build/build.sh
 fold_start() {
-	if env | grep -q -e '^TRAVIS=true$'; then
-		echo -e "travis_fold:start:$1\033[33;1m$2\033[0m"
+	if in_ci travis; then
+		echo -e "travis_fold:start:$1\033[33;1m${@:2}\033[0m"
```
Comment (purpleidea) [inline test/util.sh:61]: I'll merge this, but what's this for?

**Comment by purpleidea** [comment]
So I merged all of this. yay \o/
I snuck in one additional commit after commit #4, and I rebased #5 ... Hope that's okay.
One question about: https://github.com/purpleidea/mgmt/pull/636#discussion_r575913084
And also why did you disable the `race` tests, should we add those back in?
Lastly, it's true that some of the shell tests are flaky. I can explain, but we should just be sad about them for now.

Thank you for the patches!

**Comment by purpleidea** [comment]
PS: since github is dumb, it apparently is not smart enough to show even a partial "merged". Sorry everyone.

**Pull request closed without merging**